Library/Homebrew/diagnostic.rb

```diff
@@ -558,15 +569,34 @@ def check_brew_git_origin
         examine_git_origin(HOMEBREW_REPOSITORY, Homebrew::EnvConfig.brew_git_remote)
       end
 
-      def check_coretap_git_origin
-        examine_git_origin(CoreTap.instance.path, Homebrew::EnvConfig.core_git_remote)
+      def check_coretap_integrity
+        coretap = CoreTap.instance
+        if broken_tap?(coretap)
+          <<~EOS
+            Homebrew/homebrew-core was not tapped properly.
+            To fix:
+              rm -rf "#{coretap.path}"
+              brew tap homebrew/core
+          EOS
+        else
+          examine_git_origin(CoreTap.instance.path, Homebrew::EnvConfig.core_git_remote)
+        end
       end
 
-      def check_casktap_git_origin
+      def check_casktap_integrity
         default_cask_tap = Tap.default_cask_tap
         return unless default_cask_tap.installed?
 
-        examine_git_origin(default_cask_tap.path, default_cask_tap.remote)
+        if broken_tap?(default_cask_tap)
+          <<~EOS
+            Homebrew/homebrew-cask was not tapped properly.
+            To fix:
+              rm -rf "#{default_cask_tap.path}"
+              brew tap homebrew/cask
```

Comment:
```suggestion
            #{default_cask_tap.full_name} was not tapped properly.
            To fix:
              rm -rf "#{default_cask_tap.path}"
              brew tap #{default_cask_tap.name}
```

which also makes me wonder if `broken_tap?` (without the `?`) could return this message itself?